growattServer/__init__.py

```diff
@@ -515,3 +515,119 @@ def plant_info(self, plant_id):
+
+        Returns:
+        A response from the server stating whether the configuration was successful or not
+        """
+        #These are the parameters that the form requires, without these an error is thrown. Pre-populate their values with the current values
+        form_settings = {
+            'plantCoal': (None, str(current_settings['formulaCoal'])),
+            'plantSo2': (None, str(current_settings['formulaSo2'])),
+            'accountName': (None, str(current_settings['userAccount'])),
+            'plantID': (None, str(current_settings['id'])),
+            'plantFirm': (None, '0'), #Hardcoded to 0 as I can't work out what value it should have
+            'plantCountry': (None, str(current_settings['country'])),
+            'plantType': (None, str(current_settings['plantType'])),
+            'plantIncome': (None, str(current_settings['formulaMoneyStr'])),
+            'plantAddress': (None, str(current_settings['plantAddress'])),
+            'plantTimezone': (None, str(current_settings['timezone'])),
+            'plantLng': (None, str(current_settings['plant_lng'])),
+            'plantCity': (None, str(current_settings['city'])),
+            'plantCo2': (None, str(current_settings['formulaCo2'])),
+            'plantMoney': (None, str(current_settings['formulaMoneyUnitId'])),
+            'plantPower': (None, str(current_settings['nominalPower'])),
+            'plantLat': (None, str(current_settings['plant_lat'])),
+            'plantDate': (None, str(current_settings['createDateText'])),
+            'plantName': (None, str(current_settings['plantName'])),
+        }
+
+        #Overwrite the current value of the setting with the new value
+        for setting, value in changed_settings.items():
+            form_settings[setting] = (None, str(value))
+
+        response = self.session.post(self.get_url('newTwoPlantAPI.do?op=updatePlant'), files = form_settings)
+        data = json.loads(response.content.decode('utf-8'))
+        return data
+
+    def apply_inverter_setting_array(self, serial_number, setting_type, params_array):
+        """
+        Applies the setting using the apply_inverter_setting function, but assumes all params are numbered incrementally from param1 ..... paramN
+
+        Keyword arguments:
+        serial_number -- The serial number (device_sn) of the inverter
+        setting_type -- The setting to be configured (list of known working settings below)
+        params_array -- An array of parameters for the setting_type
+
+        Returns:
+        A response from the server stating whether the configuration was successful or not
+        """
+        params_dict = {}
+        for index, param in enumerate(params_array, start=1):
+            params_dict['param' + str(index)] = param
+
+        return self.apply_inverter_setting(serial_number, setting_type, params_dict)
+
+    def apply_inverter_setting(self, serial_number, setting_type, parameters):
```

Comment:
Same here with the magic of it, if we can detect that it's an array/list instead of a dict do it in the same function.

However i would also use different naming here, freaking confusing names 😛 
However this function does seem to be for a mix device right? if that's the case it's best for the function name to reflect that

Reply:
Magic added for array/list/dictionary handling.

Naming changed - you're correct the fact that "mix" is in the URL implies it's only for "mix" systems, so it now reflects this and I've made the names consistent so they all say "update" now as well.